src/main/java/mate/academy/service/impl/AuthenticationServiceImpl.java

```diff
@@ -0,0 +1,36 @@
+package mate.academy.service.impl;
+
+import mate.academy.exception.AuthenticationException;
+import mate.academy.exception.RegistrationException;
+import mate.academy.lib.Inject;
+import mate.academy.lib.Service;
+import mate.academy.model.User;
+import mate.academy.service.AuthenticationService;
+import mate.academy.service.UserService;
+import mate.academy.util.HashUtil;
+
+@Service
+public class AuthenticationServiceImpl implements AuthenticationService {
+    @Inject
+    private UserService userService;
+
+    @Override
+    public User login(String email, String password) throws AuthenticationException {
+        return userService.findByEmail(email)
+                .filter(user ->
+                        user.getPassword().equals(HashUtil.hashPassword(password, user.getSalt())))
+                .orElseThrow(() -> new AuthenticationException("Login or password was incorrect"));
+    }
+
+    @Override
+    public void register(String email, String password) throws RegistrationException {
```

Comment:
According to the task in README.md method `register` should return `User` object